src/Stack/Types/Config.hs

```diff
@@ -840,6 +842,8 @@ data ConfigMonoid =
     -- ^ See: 'configExtraIncludeDirs'
     ,configMonoidExtraLibDirs        :: !(Set Text)
     -- ^ See: 'configExtraLibDirs'
+    , configMonoidOverrideGccPath    :: !(First Text)
```

Comment:
I just had a quick play with this - there's a slight problem with there being no `FromJSON` instance for `Path Abs File` etc. I'd rather fix that as part of #2267 (as it is a little tangential). I'm happy to pick that one up as well once this is done though :).